### ISSUE TITLE
Problem: Too tedious to create separated c apis for all client features

### ISSUE THREAD

**Issue by yihuang** [crypto-com/chain#1121]
Separated from #1115.

We can classify c apis into two categories:
- Stateless, these are pure functions, and we should provide them with normal c APIs, they can be embedded into user's code perfectly.
  - keypair derivation
  - transaction signing
  - fee policy
  - UTxO selection algorithm implemented in a stateless way
- Stateful features, there are features involves storage state and blocking network IO.
  These APIs should be taken extra care when used in UI thread or event-driven environment.
  - Wallet syncing.
  - Transaction building and broadcasting.
  - Basically most of the features in `client-rpc`.

I propose to expose the stateful features with a common JSON API:

```rust
fn jsonrpc_call(storage_path, tendermint_rpc_url, network_id, json_request) -> json_response
```

> It might be expanded a little bit as we investigate deeper, but the number of APIs should be a handful.

**Comment by yihuang** [comment]
Python test:
```python
import json
from ctypes import *

lib = cdll.LoadLibrary('../target/debug/libcro_clib.dylib')
lib.jsonrpc_call.argtypes = (c_char_p, c_char_p, c_char, c_char_p, c_char_p, c_size_t)

storage_path = './data/node0/wallet'.encode()
wsurl = 'ws://127.0.0.1:26657/websocket'.encode()
req = json.dumps({'jsonrpc': '2.0', 'method': 'wallet_list', 'params': [], 'id': 1}).encode()
output = create_string_buffer(1024)
retcode = lib.jsonrpc_call(storage_path, wsurl, b'\xAB', req, output, len(output))
assert retcode == 0, output.value.decode()
print(json.loads(output.value))
# {'jsonrpc': '2.0', 'result': ['Default'], 'id': 1}
```
